### ISSUE TITLE
Moderated MUC does not limit voice of new users

### ISSUE THREAD

**Issue by RubenHoms** [processone/ejabberd#2997]
Environment
-----------
- ejabberd version: 19.05
- Erlang version: `Erlang/OTP 19 [erts-8.2.1] [source] [64-bit] [smp:2:2] [async-threads:10] [kernel-poll:false]`
- OS: Linux (Debian)
- Installed from: distro package (stretch-backports repo)

Bug description
---------------
Expected behavior: Creating a moderated room makes it impossible for new people joining to send messages until granted 'participant' privileges by a moderator.

Actual behavior: Creating a moderated room allows everyone to join as a 'participant', therefore giving them write access to the MUC.

As described in [XEP-0045](https://xmpp.org/extensions/xep-0045.html) for MUC's, a moderated room is: 
`A room in which only those with "voice" are allowed to send messages to all occupants.`. 

Voice is defined as: 
`In a moderated room, the privilege to send messages to all occupants.`.

So in short, someone joining a moderated room should not be able to send messages until given a voice by a moderator.

I could not find anything in the documentation about this so it seems like a bug at this point. I tried to do the same with an Openfire server and got the expected behavior as defined in XEP-0045.

**Comment by RubenHoms** [comment]
Found out what was causing this. There's a config option for `mod_muc` called `members_by_default`, which by default makes new people a participant. When you turn this off for your room new people will not have voice.
